### PR TITLE
Added Multipath, Next Hop Propagation and Set Communities options to the Action Rule Profile resource

### PR DESCRIPTION
**Pull request by sajagana**
Note:

Added Multipath, Next Hop Propagation and Set Communities options to the Action Rule Profile resource, PR belongs to: https://github.com/CiscoDevNet/terraform-provider-aci/issues/826

### CODE REVIEW

File: aci/data_source_aci_rtctrlattrp.go [hunk 1]
```diff
@@ -21,14 +23,30 @@ func dataSourceAciActionRuleProfile() *schema.Resource {
...
 				Optional: true,
-				Computed: true,
+				Type:     schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
 			},
-			"name": {
+			"next_hop_propagation": {
 				Type:     schema.TypeString,
```
Comment (lhercot) [inline aci/data_source_aci_rtctrlattrp.go:38]: In UI it is a simple boolean. Should we just make it yes or no?

File: aci/data_source_aci_rtctrlattrp.go [hunk 2]
```diff
@@ -21,14 +23,30 @@ func dataSourceAciActionRuleProfile() *schema.Resource {
...
-				Required: true,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"nh-unchanged",
+				}, false),
+			},
+			"multipath": {
```
Comment (lhercot) [inline aci/data_source_aci_rtctrlattrp.go:44]: In UI it is a simple boolean. Should we just make it yes or no?

File: aci/resource_aci_rtctrlattrp.go [hunk 3]
```diff
@@ -34,6 +35,27 @@ func resourceAciActionRuleProfile() *schema.Resource {
...
 			},
+			"set_communities": {
+				Optional: true,
+				Type:     schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"next_hop_propagation": {
+				Type:     schema.TypeString,
```
Comment (lhercot) [inline aci/resource_aci_rtctrlattrp.go:46]: In UI it is a simple boolean. Should we just make it yes or no?

File: aci/resource_aci_rtctrlattrp.go [hunk 4]
```diff
@@ -34,6 +35,27 @@ func resourceAciActionRuleProfile() *schema.Resource {
...
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"nh-unchanged",
+				}, false),
+			},
+			"multipath": {
```
Comment (lhercot) [inline aci/resource_aci_rtctrlattrp.go:52]: In UI it is a simple boolean. Should we just make it yes or no?

File: aci/data_source_aci_rtctrlattrp.go [hunk 5]
```diff
@@ -21,14 +23,30 @@ func dataSourceAciActionRuleProfile() *schema.Resource {
...
+				},
 			},
-			"name": {
+			"next_hop_propagation": {
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"nh-unchanged",
+				}, false),
```
Comment (lhercot) [inline aci/data_source_aci_rtctrlattrp.go:42]: no need for validate func as this is not used for input but for adding what is configured (data source)

File: aci/data_source_aci_rtctrlattrp.go [hunk 6]
```diff
@@ -21,14 +23,30 @@ func dataSourceAciActionRuleProfile() *schema.Resource {
...
+			},
+			"multipath": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"nh-unchanged",
```
Comment (lhercot) [inline aci/data_source_aci_rtctrlattrp.go:48]: no need for validate func as this is not used for input but for adding what is configured (data source)

File: aci/resource_aci_rtctrlattrp.go [hunk 7]
```diff
@@ -128,6 +258,62 @@ func resourceAciActionRuleProfileCreate(ctx context.Context, d *schema.ResourceD
...
+		rtctrlSetCommAttr.SetCriteria = setCommMap["criteria"]
+		rtctrlSetComm := models.NewRtctrlSetComm(fmt.Sprintf(models.RnrtctrlSetComm), rtctrlAttrP.DistinguishedName, "", "", rtctrlSetCommAttr)
+
+		creation_err := aciClient.Save(rtctrlSetComm)
+		if creation_err != nil {
+			return diag.FromErr(creation_err)
+		}
+
+		log.Printf("[DEBUG] %s: Creation finished successfully", rtctrlSetComm.DistinguishedName)
+		resourceAciRtctrlSetCommRead(ctx, rtctrlSetComm.DistinguishedName, d, m)
```
Comment (lhercot) [inline aci/resource_aci_rtctrlattrp.go:278]: Why reading here and also in the main read function called at the end? Isn't this a double read ? (See same comment in other PR and this apply to all the reads for sub-options)

File: aci/resource_aci_rtctrlattrp.go [hunk 8]
```diff
@@ -199,6 +509,60 @@ func resourceAciActionRuleProfileRead(ctx context.Context, d *schema.ResourceDat
 	return nil
 }
 
+func resourceAciRtctrlSetCommRead(ctx context.Context, dn string, d *schema.ResourceData, m interface{}) diag.Diagnostics {
```
Comment (lhercot) [inline aci/resource_aci_rtctrlattrp.go:512]: Not sure those read functions are necessary (see above comments on the double reads)

File: website/docs/d/action_rule_profile.html.markdown [hunk 9]
```diff
@@ -39,3 +39,6 @@ data "aci_action_rule_profile" "example" {
 * `annotation` - (Optional) Annotation of object Action Rule Profile.
 * `name_alias` - (Optional) Name Alias of object Action Rule Profile.
 * `description` - (Optional) Description of object Action Rule Profile.
+* `set_communities` - (Optional) Set Communities of object Action Rule Profile.
```
Comment (lhercot) [inline website/docs/d/action_rule_profile.html.markdown:42]: Document what are the expected map attributes and add type Map to the documentation.

File: website/docs/r/action_rule_profile.html.markdown [hunk 10]
```diff
@@ -37,6 +43,9 @@ resource "aci_action_rule_profile" "example" {
 * `annotation` - (Optional) Annotation of the object Action Rule Profile.
 * `description` - (Optional) Description of the object Action Rule Profile.
 * `name_alias` - (Optional) Name alias of the object Action Rule Profile.
+* `set_communities` - (Optional) Set Communities of object Action Rule Profile.
```
Comment (lhercot) [inline website/docs/r/action_rule_profile.html.markdown:46]: Document what are the expected attributes in the map and add type Map to the documentation.

**Review by lhercot — CHANGES_REQUESTED** [review]

File: aci/resource_aci_rtctrlattrp.go (hunk 7 above)
Comment (sajagana) [reply aci/resource_aci_rtctrlattrp.go:336]: As per the code, we developed a Read function call for all independent child classes and at the end of the Create or Update function we are calling the Action Rule Profile read function.

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**